obspy/io/gcf/libgcf.py

```diff
@@ -0,0 +1,159 @@
+
+
+def decode36(data):
+    """
+    Decode base 36 data
+    """
+    # http://geophysics.eas.gatech.edu/GTEQ/Scream4.4/Decoding_Base_36_numbers_C.htm
+    s = ''
+    while data:
+        imed = data % 36
+        if imed > 9:
+            imed += 7
+        s = chr(imed + 48) + s
+        data = np.uint32(data / 36)
+    return s
+
+
+def decode_date_time(data):
+    """
+    Decode date and time field.
+
+    The date code is a 32 bit value specifying the start time of the block.
+    Bits 0-16 contain the number of seconds since midnight,
+    and bits 17-31 the number of days since 17th November 1989.
+    """
+    days = data >> 17
+    secs = data & 0x1FFFF
+    starttime = UTCDateTime('1989-11-17') + days * 86400 + secs
+    return starttime
+
+
+def read_data_block(f, headonly=False, channel_prefix="HH", **kwargs):
+    """Read one data block from GCF file.
+
+    more details can be found here:
+    http://geophysics.eas.gatech.edu/GTEQ/Scream4.4/GCF_Specification.htm
+    f - file object to read from
+    if skipData is True, Only header is returned.
+    if not a data block (SPS=0) - returns None.
+    """
+    # get ID
+    sysid = f.read(4)
+    if not sysid:
+        raise EOFError  # got to EOF
+    sysid = np.frombuffer(sysid, count=1, dtype='>u4')
+    if sysid >> 31 & 0b1 > 0:
+        sysid = (sysid << 6) >> 6
+    sysid = decode36(sysid)
+    # get Stream ID
+    stid = np.frombuffer(f.read(4), count=1, dtype='>u4')
+    stid = decode36(stid)
+    # get Date & Time
+    data = np.frombuffer(f.read(4), count=1, dtype='>u4')
+    starttime = decode_date_time(data)
+    # get data format
+    # get reserved, SPS, data type compression, number of 32bit records (N)
+    reserved, sps, compress, N = np.frombuffer(f.read(4), count=4, dtype='>u1')
+    compression = compress & 0b00001111  # get compression code
+    t_offset = compress >> 4  # get time offset
+    if t_offset > 0:
+        starttime = starttime + t_offset / time_offsets_d[sps]
+    if sps in sps_d:
+        sps = sps_d[sps]  # get special SPS value if needed
+    if not sps:
+        f.seek(N * 4, 1)  # skip if not a data block
+        if 1008 - N * 4 > 0:
+            f.seek(1008 - N * 4, 1)  # keep skipping to get 1008 record
+        return None
+    npts = int(N) * compression  # number of samples
+    header = {}
+    header['starttime'] = starttime
+    header['station'] = stid[:4]
+    header['channel'] = (channel_prefix[:2] + stid[4]).upper()
+    header['sampling_rate'] = float(sps)
+    header['npts'] = npts
+    if headonly:
+        f.seek(4 * (N + 2), 1)  # skip data part (inc. FIC and RIC)
+        # skip to end of block if only partly filled with data
+        if 1000 - int(N) * 4 > 0:
+            f.seek(1000 - int(N) * 4, 1)
+        return header
+    else:
+        # get FIC
+        fic = np.frombuffer(f.read(4), count=1, dtype='>i4')
+        # get incremental data
+        data = np.frombuffer(f.read(4 * N), count=npts,
```

Comment:
N is the number of 4 bytes records with data, but the data could be arranged as 1, 2, or 4 bytes. You can have 200 samples of 16bits and N would be 100. It was easier for me to write it this way without getting confused. I think its easier since it's always 32 bits to read and no need to account for dtype.